chichiwang/exercise-js
# Move `client/app.tsx` to `app/index.tsx`

## Description
Linked to Issue: #17 
Move `client/app.tsx` to `app/index.tsx`, update associated mappings and snapshot tests. Rationale: the `client/` will contain the client bundle entrypoint (and any necessary client bundle scripts), the `app/` directory will contain all top level application concerns (base markup, hooks, routing, etc).

## Changes
* File `client/app.tsx` has been moved to `app/index.tsx` along with tests
* `.eslintrc.js` has had mapping configurations updated to map to the `app/` directory
* `jest.config.json` has had mapping configurations updated to map to the `app/` directory
* `tsconfig.json` has had mapping configurations updated to map to the `app/` directory
* `webpack-config.js` has had mapping configurations updated to map to the `app/` directory
* `webpack-config.js` has been configured to also watch the `app/` directory
* `server/index.tsx` has been updated to import the `App` component from the `app/` directory

## Steps to QA
* Pull down and switch to this branch
* Run `npm run build:dev` to ensure the application still builds correctly
  * Run `node dist/index.js` to verify the server still starts correctly. Verify in browser that the app runs correctly
* Run `npm start` to ensure the watcher runs correctly
  * Verify in browser
  * Make a change in `server/`, `app/`, and `client/` files to ensure the watch process rebuilds the app on change
* Ensure passing linter and tests
